x/cdp/simulation/genesis.go

```diff
@@ -4,19 +4,164 @@ import (
 	"fmt"
 
 	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
-
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	authexported "github.com/cosmos/cosmos-sdk/x/auth/exported"
+	"github.com/cosmos/cosmos-sdk/x/supply"
+	supplyExported "github.com/cosmos/cosmos-sdk/x/supply/exported"
 	"github.com/kava-labs/kava/x/cdp/types"
 )
 
 // RandomizedGenState generates a random GenesisState for cdp
 func RandomizedGenState(simState *module.SimulationState) {
 
-	// TODO implement this fully
-	// - randomly generating the genesis params
-	// - overwriting with genesis provided to simulation
-	cdpGenesis := types.DefaultGenesisState()
+	cdpGenesis := randomCdpGenState(simState.Rand.Intn(2))
+
+	// hacky way to give accounts coins so they can create cdps (coins  includes usdx so it's possible to have sufficient balance to close a cdp)
+	var authGenesis auth.GenesisState
+	simState.Cdc.MustUnmarshalJSON(simState.GenState[auth.ModuleName], &authGenesis)
+	totalCdpCoins := sdk.NewCoins()
+	for _, acc := range authGenesis.Accounts {
+		_, ok := acc.(supplyExported.ModuleAccountI)
+		if ok {
+			continue
+		}
+		coinsToAdd := sdk.NewCoins(
+			sdk.NewCoin("bnb", sdk.NewInt(int64(simState.Rand.Intn(100000000000)))),
+			sdk.NewCoin("xrp", sdk.NewInt(int64(simState.Rand.Intn(100000000000)))),
+			sdk.NewCoin("btc", sdk.NewInt(int64(simState.Rand.Intn(500000000)))),
+			sdk.NewCoin("usdx", sdk.NewInt(int64(simState.Rand.Intn(1000000000)))),
+		)
+		err := acc.SetCoins(acc.GetCoins().Add(coinsToAdd))
+		if err != nil {
+			panic(err)
+		}
+		totalCdpCoins = totalCdpCoins.Add(coinsToAdd)
+		authGenesis.Accounts = replaceOrAppendAccount(authGenesis.Accounts, acc)
+	}
+	simState.GenState[auth.ModuleName] = simState.Cdc.MustMarshalJSON(authGenesis)
+
+	var supplyGenesis supply.GenesisState
+	simState.Cdc.MustUnmarshalJSON(simState.GenState[supply.ModuleName], &supplyGenesis)
+	supplyGenesis.Supply = supplyGenesis.Supply.Add(totalCdpCoins)
+	simState.GenState[supply.ModuleName] = simState.Cdc.MustMarshalJSON(supplyGenesis)
 
 	fmt.Printf("Selected randomly generated %s parameters:\n%s\n", types.ModuleName, codec.MustMarshalJSONIndent(simState.Cdc, cdpGenesis))
 	simState.GenState[types.ModuleName] = simState.Cdc.MustMarshalJSON(cdpGenesis)
 }
+
+// In a list of accounts, replace the first account found with the same address. If not found, append the account.
+func replaceOrAppendAccount(accounts []authexported.GenesisAccount, acc authexported.GenesisAccount) []authexported.GenesisAccount {
+	newAccounts := accounts
+	for i, a := range accounts {
+		if a.GetAddress().Equals(acc.GetAddress()) {
+			newAccounts[i] = acc
+			return newAccounts
+		}
+	}
+	return append(newAccounts, acc)
+}
+
+func randomCdpGenState(selection int) types.GenesisState {
```

Comment:
I guess is ok for now to have the genesis state hardcoded. In the future, we might want to make it actually random with different collateral params, thresholds, etc.